### PR TITLE
Content directory runtime integration

### PR DESCRIPTION
**Pull request by iorveth**
The scope of this PR includes:
1. `content_directory` module runtime integration 
1. `content_directory `and `working_group` & `membership` modules integration related work, implemented through `content_directory::ActorAuthenticator ` trait.

Closes https://github.com/Joystream/joystream/issues/1281, https://github.com/Joystream/joystream/issues/1275

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -372,6 +377,59 @@ impl versioned_store_permissions::Trait for Runtime {
...
 }
 
+type EntityId = <Runtime as content_directory::Trait>::EntityId;
+
+parameter_types! {
+    pub const PropertyNameLengthConstraint: InputValidationLengthConstraint = InputValidationLengthConstraint::new(1, 49);
+    pub const PropertyDescriptionLengthConstraint: InputValidationLengthConstraint = InputValidationLengthConstraint::new(1, 500);
+    pub const ClassNameLengthConstraint: InputValidationLengthConstraint = InputValidationLengthConstraint::new(1, 49);
+    pub const ClassDescriptionLengthConstraint: InputValidationLengthConstraint = InputValidationLengthConstraint::new(1, 500);
+
```
Comment (shamil-gadelshin) [inline runtime/src/lib.rs]: Do we need spaces between lines?

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -372,6 +377,59 @@ impl versioned_store_permissions::Trait for Runtime {
...
+    pub const IndividualEntitiesCreationLimit: EntityId = 50;
+}
+
+impl content_directory::Trait for Runtime {
+    type Event = Event;
+    type Nonce = u64;
+    type ClassId = u64;
+    type EntityId = u64;
+
```
Comment (shamil-gadelshin) [inline runtime/src/lib.rs]: Do we need spaces between lines?

File: runtime-modules/content-directory/src/permissions.rs [hunk 3]
```diff
@@ -122,7 +121,6 @@ impl<T: Trait> Default for Actor<T> {
 }
 
 impl<T: Trait> core::fmt::Debug for Actor<T> {
-    #[cfg(feature = "std")]
     fn fmt(&self, formatter: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
```
Comment (shamil-gadelshin) [inline runtime-modules/content-directory/src/permissions.rs:124]: Why did you remove conditional compilation for this and other Debug implementations>

File: runtime-modules/working-group/src/lib.rs [hunk 4]
```diff
@@ -1298,7 +1298,7 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
         Ok(worker)
     }
```
Comment (shamil-gadelshin) [inline runtime-modules/working-group/src/lib.rs:1300]: Add a comment for a public method, please.

File: runtime/src/integration/content_directory.rs [hunk 5]
```diff
@@ -0,0 +1,47 @@
+use crate::{AccountId, MemberId, Runtime};
+
+// The content directory working group instance alias.
```
Comment (shamil-gadelshin) [inline runtime/src/integration/content_directory.rs]: It is already taken. Take instance3 instead. Also to prevent future errors - please move this line to the lib.rs closer to the Instance2 declaration.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
We need to implement the working_group for Instance3, check StorageWorkingGroupInstance implementation for details.

**Review by bedeho — APPROVED** [review]
LGTM.

File: runtime-modules/content-directory/src/permissions.rs (hunk 3 above)
Comment (iorveth) [reply runtime-modules/content-directory/src/permissions.rs:124]: I still getting errors, related to T: Trait inheritance, when conditional compilation enabled here. I hope this will be fixed with further refactoring.

File: runtime/src/integration/content_directory.rs (hunk 5 above)
Comment (iorveth) [reply runtime/src/integration/content_directory.rs]: Ok. But i saw, both service-discovery and storage modules are using `Instance2` Is that correct case?

Comment (iorveth) [reply runtime/src/integration/content_directory.rs]: Done here https://github.com/Joystream/joystream/pull/1273/commits/b9c741ed8b0adecd6ee1b13e67fba8604574e787

File: runtime-modules/working-group/src/lib.rs (hunk 4 above)
Comment (iorveth) [reply runtime-modules/working-group/src/lib.rs:1300]: Done https://github.com/Joystream/joystream/pull/1273/commits/9d6bbb08706503ef06060959476baf0028b1fdf8

File: runtime/src/lib.rs (hunk 1 above)
Comment (iorveth) [reply runtime/src/lib.rs]: The purpose is to make constraints of different types distinguishable. I can remove them, though.

File: runtime/src/lib.rs (hunk 2 above)
Comment (iorveth) [reply runtime/src/lib.rs]: The purpose is to make constraints of different types distinguishable. I can remove them, though.

File: runtime-modules/content-directory/src/permissions.rs (hunk 3 above)
Comment (shamil-gadelshin) [reply runtime-modules/content-directory/src/permissions.rs:124]: Ok. Please, keep in mind to remove those implemented traits after the refactoring.

File: runtime/src/integration/content_directory.rs (hunk 5 above)
Comment (shamil-gadelshin) [reply runtime/src/integration/content_directory.rs]: Yes. Because they use the same working group.

File: runtime/src/lib.rs (hunk 1 above)
Comment (shamil-gadelshin) [reply runtime/src/lib.rs]: Yes, please. Remove or introduce explicit comments for different types.

Comment (iorveth) [reply runtime/src/lib.rs]: Done. Removed them in content directory `mock.rs` as well. https://github.com/Joystream/joystream/pull/1273/commits/d993a64a39c491aad16e4c3d0f1b041496d41f37

**Review by shamil-gadelshin — APPROVED** [review]
LGTM
